pgx/hex.py

```diff
@@ -110,57 +130,37 @@ def merge(i, b):
         )
 
     board = jax.lax.fori_loop(0, 6, merge, board)
-    won = _is_game_end(board, size, state._x.turn)
-    reward = jax.lax.cond(
-        won,
-        lambda: jnp.float32([-1, -1]).at[state.current_player].set(1),
-        lambda: jnp.zeros(2, jnp.float32),
+    return state._replace(
+        step_count=state.step_count + 1,
+        board=board * -1,
     )
 
-    state = state.replace(  # type:ignore
-        current_player=1 - state.current_player,
-        _x=GameState(
-            turn=1 - state._x.turn,
-            board=board * -1,
-        ),
-        rewards=reward,
-        terminated=won,
-        legal_action_mask=state.legal_action_mask.at[:-1].set(board == 0).at[-1].set(state._step_count == 1),
-    )
-
-    return state
 
-
-def _swap(state: State, size: int) -> State:
-    ix = jnp.nonzero(state._x.board, size=1)[0]
+def _swap(state: GameState, size: int) -> GameState:
+    ix = jnp.nonzero(state.board, size=1)[0]
     row = ix // size
     col = ix % size
     swapped_ix = col * size + row
     set_place_id = swapped_ix + 1
-    board = state._x.board.at[ix].set(0).at[swapped_ix].set(set_place_id)
-    return state.replace(  # type: ignore
-        current_player=1 - state.current_player,
-        _x=GameState(
-            turn=1 - state._x.turn,
-            board=board * -1,
-        ),
-        legal_action_mask=state.legal_action_mask.at[:-1].set(board == 0).at[-1].set(FALSE),
+    board = state.board.at[ix].set(0).at[swapped_ix].set(set_place_id)
+    return state._replace(
+        step_count=state.step_count + 1,
+        board=board * -1,
     )
 
 
-def _observe(state: State, player_id: Array, size) -> Array:
-    board = jax.lax.select(
-        player_id == state.current_player,
-        state._x.board.reshape((size, size)),
-        -state._x.board.reshape((size, size)),
-    )
+def _observe(state: GameState, color: Optional[Array] = None, size: int = 11) -> Array:
+    if color is None:
+        color = state.color
+
+    board = jax.lax.select(color == state.color, state.board, -state.board)
+    board = board.reshape((size, size))
 
     my_board = board * 1 > 0
     opp_board = board * -1 > 0
     ones = jnp.ones_like(my_board)
-    color = jax.lax.select(player_id == state.current_player, state._x.turn, 1 - state._x.turn)
     color = color * ones
-    can_swap = state.legal_action_mask[-1] * ones
+    can_swap = (state.step_count == 1) * ones
```

Comment:
This line changes the observation in *terminated* state.
`can_swap` plane changes from True to False.